mon-pix/app/components/signin-form.js

```diff
@@ -34,44 +34,58 @@ export default class SigninForm extends Component {
     this.hasFailed = false;
     try {
       await this.session.authenticateUser(this.login, this.password);
-    } catch (response) {
-      const shouldChangePassword = get(response, 'responseJSON.errors[0].title') === 'PasswordShouldChange';
-      if (shouldChangePassword) {
-        const passwordResetToken = response.responseJSON.errors[0].meta;
-        await this._updateExpiredPassword(passwordResetToken);
-      } else {
-        this.errorMessage = this._findErrorMessage(get(response, 'responseJSON.errors[0].code'), response.status);
-      }
+    } catch (responseError) {
       this.hasFailed = true;
+      this._handleApiError(responseError);
     }
   }
 
-  _findErrorMessage(errorCode, statusCode) {
-    let httpStatusCodeMessages;
-
-    switch (errorCode) {
+  async _handleApiError(responseError) {
+    const errors = get(responseError, 'responseJSON.errors');
+    let error;
+    if (Array.isArray(errors) && errors.length > 0) {
+      error = errors[0];
+    }
```

Comment:
```suggestion
const error = (Array.isArray(errors) && errors.length >0) && errors[0];
```
(Pour éviter de déclarer avec un `let`)

Reply:
@Anne-Gaelle-S ta remarque me montre que je n'ai pas été au bout de mon idée en fait. Merci. Je vais faire une modification à ce niveau.

Reply:
Modification faite suivant la recommandation.